File: Backend/src/main/java/org/example/backend/domain/Teacher.java
package org.example.backend.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "teacher")
public class Teacher {
    @Id
    @Column(name = "id", nullable = false, length = 6)
    private String id;

    @MapsId
    @OneToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "id", nullable = false)
    private User user;

    @Column(name = "job_title", nullable = false, length = 20)
    private String jobTitle;

    public Teacher() {
    }

    public Teacher(String id) {
        this.id = id;
    }
}

File: TTSolver/Model/Teacher.py
class Teacher:
    def __init__(self, id: str, user: 'User', job_title: str):
        self.id = id
        self.user = user
        self.job_title = job_title

    def __str__(self):
        return f"Teacher(id={self.id}, user={self.user}, job_title={self.job_title})"
